### PR TITLE
cleanup round 2

### PR DESCRIPTION
**Pull request by h0uter**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -69,13 +67,25 @@ pub fn run(config: &mut Config) -> Result<(), Box<dyn Error>> {
...
+    pub fn build() -> Config {
+        let secret_word = get_user_input("Provide your secret word:");
+        let correctly_guessed_letters = "_".repeat(secret_word.len());
+
+        return Config {
+            secret_word,
+            correctly_guessed_letters,
+        };
+    }
 }
```
Comment (bobluppes) [inline src/lib.rs:84]: Nice that you introduced such an build method  again 👍🏻 

Seeing as `correctly_guessed_letters` depends on the `secret_word`, I think it is indeed nice to create it in such a constructor. However, I could also see the argument for extracting (part of) this build logic into a free function. 

At the end of the day I think we are both trying to adhere to the SRP. While I agree that constructing the `correctly_guessed_letters` should be the responsibility of the config class, I wonder whether it makes sense to extract the io and ask for the user input in a free function.

What do you think?

**Review by bobluppes — APPROVED** [review]
LGTM let's gooo

I am interested in your opinion on the comment below, but feel free to merge already.

File: src/lib.rs (hunk 1 above)
Comment (h0uter) [reply src/lib.rs:84]: For educational purposes I would like to set it up in such a way that we can pass the `secret_word` as a CLI argument, but if it is not passed then prompt the user for it.

I also feel like we should split config (static) and gamestate (transient). Where `secret_word` should be part of config and the `correctly_guessed_letters` should be part of GameState.

Comment (bobluppes) [reply src/lib.rs:84]: 👍🏻 nice, I like both your proposals! Especially the difference between static and transient context is a great one.

Regarding the initial point on extracting the io from the `Config`, I think I would need some more experience in rust to figure out what the idiomatic way is 😅 . In the end it comes down to whether this `build` method is considered part of the interface of the struct, or whether it is seen as a sort of external factory.

**Pull request merged**